dynawo/sources/Models/CPP/Controls/Voltage/ModelVoltageMeasurementsUtilities/DYNModelVoltageMeasurementsUtilities.h

```diff
@@ -0,0 +1,375 @@
+   */
+  void evalDynamicFType() { /* not needed */}
+
+  /**
+   * @copydoc ModelCPP::getY0()
+   */
+  void getY0();
+
+  /**
+   * @copydoc ModelCPP::evalStaticYType()
+   */
+  void evalStaticYType();
+
+  /**
+   * @copydoc ModelCPP::evalDynamicYType()
+   */
+  void evalDynamicYType() { /* not needed */}
+
+  /**
+   * @brief get the index of variables used to define the jacobian associated to a calculated variable
+   *
+   * @param iCalculatedVar index of the calculated variable
+   * @param indexes vector to fill with the indexes
+   */
+  void getIndexesOfVariablesUsedForCalculatedVarI(unsigned iCalculatedVar, std::vector<int>& indexes) const;
+
+  /**
+   * @brief evaluate the jacobian associated to a calculated variable
+   *
+   * @param iCalculatedVar index of the calculated variable
+   * @param res values of the jacobian
+   */
+  void evalJCalculatedVarI(unsigned iCalculatedVar, std::vector<double>& res)const;
+
+  /**
+   * @brief evaluate the value of a calculated variable
+   *
+   * @param iCalculatedVar index of the calculated variable
+   *
+   * @return value of the calculated variable
+   */
+  double evalCalculatedVarI(unsigned iCalculatedVar) const;
+
+  /**
+   * @brief ModelVoltageMeasurementsUtilities parameters setter
+   */
+  void setSubModelParameters();
+
+  /**
+   * @brief ModelVoltageMeasurementsUtilities elements initializer
+   *
+   * Define elements for this model( elements to be seen by other models)
+   *
+   * @param elements  Reference to elements' vector
+   * @param mapElement Map associating each element index in the elements vector to its name
+   */
+  void defineElements(std::vector<Element> &elements, std::map<std::string, int>& mapElement);
+
+  /**
+   * @copydoc SubModel::dumpUserReadableElementList()
+   */
+  void dumpUserReadableElementList(const std::string& nameElement) const;
+
+  /**
+   * @brief initialize variables of the model
+   *
+   * A variable is a structure which contains all information needed to interact with the model
+   * @param variables vector to fill with each variables
+   */
+  void defineVariables(std::vector<boost::shared_ptr<Variable> >& variables);
+
+  /**
+   * @brief define parameters
+   * @param parameters vector to fill with each parameters
+   */
+  void defineParameters(std::vector<ParameterModeler>& parameters);
+
+  /**
+   * @brief get check sum number
+   * @return the check sum number associated to the model
+   */
+  std::string getCheckSum() const;
+
+  /**
+   * @copydoc ModelCPP::initializeStaticData()
+   */
+  void initializeStaticData() { /* not needed */ }
+
+  /**
+   * @copydoc ModelCPP::initializeFromData(const boost::shared_ptr<DataInterface> &data)
+   */
+  void initializeFromData(const boost::shared_ptr<DataInterface>& data);
+
+  /**
+   * @copydoc ModelCPP::setFequations()
+   */
+  void setFequations();
+
+  /**
+   * @copydoc ModelCPP::setGequations()
+   */
+  void setGequations() { /* not needed */ }
```

Comment:
We do have a root so this one should be filled

Reply:
Should be okay now